Fixes #19436 - Better controller selection on host dropdown

tbrisker opened the pull request:
Always setting to `host` unless the controller is `hostgroup` broke
discovery's usage of the form as it uses the `discoverd_host`
controller.

tbrisker committed:
Fixes #19436 - Better controller selection on host dropdown

Always setting to `host` unless the controller is `hostgroup` broke
discovery's usage of the form as it uses the `discoverd_host`
controller.
This reverts the change from `method_path` to `host_or_hostgroup_path`
from commit 637da2f28accc1378f0326d97adcbb2efc5d9808 and moves the
controller correction into `method_path`.

timogoebel commented:
@tbrisker : Somehow I don't like the approach either. I'm curious if we can find a way that generates the correct url based on the same criteria for
a) the form action
b) the ajax data

tbrisker commented:
Since some of the dropdowns were using `method_path` while others had `host_or_hostgroup_path`, I reverted the change to `host_or_hostgroup_path` from 637da2f and moved the controller correction to the `method_path` method. Another option would be to override this method only in the compute_resource_vms controller.
While I agree this is a bit of a hack and would be happy to clean up the host form, this bug is a regression introduced in 1.15 which breaks discovery.

timogoebel commented:
@tbrisker: 👍  for 1.15 and 👍 for agreeing, that this isn't great and needs more work for 1.16.

dLobatog commented:
Thanks @tbrisker , setting back to WoC as it's failing tests. [test katello] (I'm not so sure katello failures happened after this)

dLobatog reviewed (changes requested):
Looks like architecture_selected_hosts_path isn't working

lzap commented:
@dLobatog does 1.15 mean RCx or 1.15.1?

tbrisker commented:
@dLobatog - thanks, I had a typo. fixed now.
@lzap - I believe this is a 1.15.0 blocker.

dLobatog commented:
[test upgrade]  - @tbrisker MySQL upgrade job failed, I will retry and merge if green

dLobatog commented:
Thanks @tbrisker !
